src/pylint_datetime/__init__.py

```diff
@@ -1,4 +1,185 @@
+        ),
+        "W9996": (
+            'Attribute access "%s" should be followed by a call to replace with timezone argument',
+            "naive-datetime-no-timezone",
+            'Attribute access "%s" should be followed by a call to replace with timezone argument',
+        ),
+    }
+
+    def __init__(self, linter: pylint.lint.PyLinter = None) -> None:
+        """Initialize checker.
+
+        Args:
+            linter: instance of pylint linter.
+        """
+        super().__init__(linter)
+
+    def timedelta_with_keywords(
+        self, node: astroid.node_classes.NodeNG, name: str
+    ) -> None:
+        """Check calls to timedelta() and datetime.timedelta() for keyword arguments.
+
+        Args:
+            node: the node that visit_call originates from.
+            name: name of the function being called.
+        """
+        if name == "timedelta":
+            for _ in node.args:
+                self.add_message("timedelta-no-keyword-args", node=node)
+            if not node.args and not node.keywords:
+                self.add_message("timedelta-no-keyword-args", node=node)
+
+    # check for positional arguments cheesy, uses fact, that they have one Attribute Argument.
+
+    def timezone_argument(self, node: astroid.node_classes.NodeNG, name: str) -> None:
+        """Check calls to datetime, now, fromtimestamp, astimezone and time for timezone arg.
+
+        Args:
+            node: the node that visit_call originates from.
+            name: name of the function being called.
+        """
+        if name in ("datetime", "now", "fromtimestamp", "astimezone"):
+            if not any(
+                isinstance(arg, astroid.Keyword) and arg.arg in ("tz", "tzinfo")
+                for arg in node.keywords
+            ):
+                if not any(isinstance(arg, astroid.Attribute) for arg in node.args):
+                    self.add_message("timezone-no-argument", node=node, args=(name,))
+
+    def naive_functions(self, node: astroid.node_classes.NodeNG, name: str) -> None:
+        """Check that functions producing only naive objects are never called.
+
+        Args:
+            node: the node that visit_call originates from
+            name: name of the function being called
+        """
+        if name in ("today", "utcnow", "utcfromtimestamp", "utctimetuple", "time"):
+            self.add_message("naive-datetime-call", node=node, args=(name,))
+
+    def visit_call(self, node: astroid.node_classes.NodeNG) -> None:
+        """Pylint function responds when any function is called.
+
+        Args:
+            node: origin node for function call.
+        """
+        if hasattr(node.func, "name"):
+            name = node.func.name
+        elif hasattr(node.func, "attrname"):
+            name = node.func.attrname
+
+        self.timedelta_with_keywords(node, name)
+
+        self.timezone_argument(node, name)
+
+        self.naive_functions(node, name)
+
+    def naive_properties_methods_replace(
```

Comment:
Since this is checked only in assignments, I guess it doesn't detect stuff like `f(datetime.datetime.min)`, right?

We don't need to cover all possible uses in the first PR though, just asking to know and possibly create issues for future improvements.

Reply:
That is true. But there are other visit_ functions for pylint checkers, so that addition shouldn't be to difficult (I hope)

Reply:
Ah, cool, I see you could plug it to the attribute access, nice!